File: frontend/src/types/campaign.ts
export type CampaignStatus = 'active' | 'paused' | 'completed'

export interface Campaign {
  id: string
  title: string
  budget: number
  initial_budget: number
  start_date: string
  end_date: string
  status: CampaignStatus
  created_at: string
  updated_at: string
}

export interface CreateCampaignRequest {
  title: string
  budget: number
  start_date: string
  end_date: string
  status?: CampaignStatus
}

export interface UpdateCampaignRequest {
  title?: string
  budget?: number
  start_date?: string
  end_date?: string
  status?: CampaignStatus
}

export interface CampaignStats {
  total_impressions: number
  spent_budget: number
  remaining_budget: number
}

File: backend/internal/model/campaign.go
package model

import "time"

type CampaignStatus string

const (
	StatusActive    CampaignStatus = "active"
	StatusPaused    CampaignStatus = "paused"
	StatusCompleted CampaignStatus = "completed"
)

type Campaign struct {
	ID        string
	Title     string
	Budget    int
	StartDate time.Time
	EndDate   time.Time
	Status    CampaignStatus
	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt *time.Time
}
